xapian-core/cluster/cluster.cc

```diff
@@ -296,10 +296,10 @@ Point::Point(const TermListGroup &tlg, const Document &document_)
 	string term = *it;
 	double termfreq = tlg.get_termfreq(term);
 
-	/** If term is not a stemmed term or the term indexes only one
-	 *  one document, do not compute TF-IDF scores for them
-	 */
-	if (wdf < 1 || term[0] != 'Z' || termfreq <= 1)
+	/// If term frequency of the term is lesser than or equal
+	/// to 1 or wdf is lesser than 1, we do not consider this
+	/// term for document vector calculations
+	if (wdf < 1 || termfreq <= 1)
 	    continue;
```

Comment:
Yes I'll update the comment accordingly and do a fixup commit.
`I thought we were going to leave the hardcoded term name check for now?`
I had made this change because I realized the test for cosine distance were broken. My idea was that document termlists contain stemmed as well as unstemmed terms and we could discard the unstemmed terms by checking the first letter. But when I corrected the test, I realized that the documents in the MSet do not contain stemmed words. I wasn't sure as to why so I decided to remove that check for now and take it up separately as a part of dimensionality reduction once everything else is good.